### DISCUSSION TITLE
After update WIKIJS doesnt work anymore

### DISCUSSION THREAD

**Discussion by SirSeba98** [requarks/wiki#6645] [Help / Questions]
Hello everyone, I run Wikijs on docker.
I tried to update it to the latest version and now I get the error "[MASTER] error: Package subpath './public/extractFiles' is not defined by "exports" in /wiki/node_modules/extract-files/package.json".

As far as I know, the docker image delivers nodejs . But  I think it delivers a wrong version.

**Comment by NGPixel** [comment]
Are you using the official docker image?

**Reply by InSelfControll** [reply]
I'm having the same issue and using the official image and still having this issue.
After reverting to version 2.4 all started works again.

**Reply by InSelfControll** [reply]
How can I fix it and do a complete upgrade of wikijs and postgres

**Comment by SirSeba98** [comment]
> Are you using the official docker image?

Yes, I am using the official docker image.
After upgrading postgresql to the latest version, everything seems to work again.

I used to run postgresql on version 11.

I didn't really expected that the postgresql version could be the issue, as the error is common to appear if the nodejs version is not support by the application.

**Reply by NGPixel** [reply]
The error is completely unrelated to the db so you had something else going on.

**Comment by nicedevil007** [comment]
Same here, but I figured something out.

I reverted back to the latest version that was working on my site. It is version **2.5.299**.
Then I noticed this line in the logs:

```python
(node:1) [DEP0148] DeprecationWarning: Use of deprecated folder mapping "./public/" in the "exports" field module resolution of the package at /wiki/node_modules/extract-files/package.json.
Update this package.json to use a subpath pattern like "./public/*".
(Use `node --trace-deprecation ...` to show where the warning was **created)**
```

So how can we fix this before doing the upgrade?

EDIT => OFFTOPIC:
Also is there a guide on how to upgrade postgres to version 15 from 11 if we have installed wikijs with the old version of postgres?

**Reply by NGPixel** [reply]
The db has absolutely nothing to do with this error.

**Reply by nicedevil007** [reply]
yeah that is true, but I just want to know how we can do an upgrade for this part :) My question wasn't belonging to my lines above :P

**Comment by Xav-Pe** [comment]
Same error here.
Downgrading to  requarks/wiki:2.5.299 fixes the issue too.

I'm only using the official images on debian 12 (the problem was also present on debian 11). Docker up to date.

Same deprecation warning:

```
[DEP0148] DeprecationWarning: Use of deprecated folder mapping "./public/" in the "exports" field module resolution of the package at /wiki/node_modules/extract-files/package.json.

Update this package.json to use a subpath pattern like "./public/*"
```

**Comment by Tedbudd3** [comment]
Anyone has found a solution for this issue please? I am having the same issue after upgrading nodejs from version 16.20 to the 18 on the wiki.js, noticed this same issue when running node server ps, Package subpath './public/extractFiles' is not defined by "exports" in /wiki/node_modules/extract-files/package.json".
